packages/react-storage/src/components/StorageBrowser/Views/Controls/Download.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { downloadData } from 'aws-amplify/storage';
+
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+
+const { Button: ButtonElement, Icon: IconElement } = StorageBrowserElements;
+
+interface DownloadActionInput {
+  key: string;
+}
+
+interface DownloadActionResult {
+  key: string;
+}
+
+function downloadAction(
+  _: DownloadActionResult,
+  { key: path }: DownloadActionInput
+): Promise<DownloadActionResult> {
+  downloadData({ path });
+  return Promise.resolve({ key: path });
+}
+
+const BLOCK_NAME = `${CLASS_BASE}__download`;
+
+export interface DownloadControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> {
+  (): React.JSX.Element;
+  Button: T['Button'];
+  Icon: T['Icon'];
+}
+
+const DownloadIcon: typeof IconElement = React.forwardRef(
+  function DownloadIcon(props, ref) {
+    return (
+      <IconElement
+        {...props}
+        ref={ref}
+        className={`${BLOCK_NAME}__icon`}
+        variant="download"
+      />
+    );
+  }
+);
+
+const DownloadButton: typeof ButtonElement = React.forwardRef(
+  function DownloadButton(props, ref) {
+    return (
+      <ButtonElement
+        {...props}
+        ref={ref}
+        className={BLOCK_NAME}
+        aria-label="Download item"
+        onClick={() => {
+          downloadAction(undefined as unknown as DownloadActionResult, {
+            key: '',
```

Comment:
What do we need here in order to make this operational?

Reply:
This needs the key, bucket, and locationCredentialProvider. I updated the control so it fetches bucket/credentials and takes the key as a prop. I can update this when we integrate it into the table but it is functional for now